src/components/Navigation/Navigation.tsx

```diff
@@ -2,16 +2,16 @@ import React, { FC } from 'react';
 import { Link } from 'react-router-dom';
 import { AppBar, Toolbar, Typography } from '@material-ui/core';
 
-import { useAppStyles } from './Navigation.style';
+import { useStyles } from './Navigation.style';
 
 const Navigation: FC = ({ children }) => {
-  const classes = useAppStyles();
+  const classes = useStyles({});
```

Comment:
I don't think we want to share styles too often as well. In react we should be designing shareable components instead of thinking in shareable styles in css class names. So in my mind `useStyles` is totally local to this component only.

I also do not see the point of `Navigation.style` file at this point of time, seems like a premature optimisation. And also considering we are not taking values from a global theme object or changing style based on props, useStyles here is way too premature. An inline style would have lesser redirection.

Anyways the change here that I want to make is actually passing in an empty props to make ts happy, I don't really mind changing the name, but this is an opportunity to talk more about this pattern.

Reply:
Yeah i agree with that, thought about it also for a while. Most of time styles are heavily stick to its corresponding component anyway.

I just got used to it on my previous company were we heavily style our components, more of just a naming scheme to tell wether this is a component or a styling component because of emotion. Sorry didn't quite get the part of inline style would have lesser redirection. Personally i never been a fan of inline styles even on the normal html element, for me it clutters the element making it harder to read. I always try to segment the DOM, styling and logic JS as much as possible.

yeah i saw that also, i wondered why you put an empty object, ts is happy on my side w/o the empty prop object lol let's merge it, talk about it maybe later on when in need to create a big component with wireframes, and personalized design, so far im just thinking that this is not a really big deal as this component is really small

Reply:
> Sorry didn't quite get the part of inline style would have lesser redirection.

What I mean by that is you don't need to make multiple jumps to see the full picture. Common criminal is redux, jumping through actions, reducer, connect, saga just to visualise one feature.

```
// "lesser redirection"
const styles = {
  header: {
	backgroundColor: 'red'
  }
}

const Header = props =>	
  <div style={styles.header}>
	...
  </div>
```

Reply:
> What I mean by that is you don't need to make multiple jumps to see the full picture. Common criminal is redux, jumping through actions, reducer, connect, saga just to visualise one feature.

oh right right lol yeah that's a pain in the ass. I don't mind if we'll stick to styling in a component, as along as it's not inline direct like `<div style={{ color: 'blue }} />` something like that.